apps/omg_watcher/test/integration/invalid_exit_test.exs

```diff
@@ -0,0 +1,159 @@
+# Copyright 2018 OmiseGO Pte Ltd
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+defmodule OMG.Watcher.Integration.InvalidExitTest do
+  use ExUnitFixtures
+  use ExUnit.Case, async: false
+  use OMG.API.Fixtures
+  use OMG.API.Integration.Fixtures
+  use Plug.Test
+  use Phoenix.ChannelTest
+
+  alias OMG.API
+  alias OMG.Eth
+  alias OMG.JSONRPC.Client
+  alias OMG.Watcher.Eventer.Event
+  alias OMG.Watcher.Integration.TestHelper, as: IntegrationTest
+  alias OMG.Watcher.Web.Channel
+
+  @moduletag :integration
+
+  @timeout 40_000
+  @eth API.Crypto.zero_address()
+
+  @endpoint OMG.Watcher.Web.Endpoint
+
+  @tag fixtures: [:watcher_sandbox, :stable_alice, :child_chain, :token, :stable_alice_deposits]
+  test "transaction which is using already spent utxo from exit and happened before end of m_sv causes to emit invalid_exit event ",
+       %{stable_alice: alice, stable_alice_deposits: {deposit_blknum, _}} do
+    # TODO remove this tx , use directly deposit_blknum to get_exit_data
+    tx = API.TestHelper.create_encoded([{deposit_blknum, 0, 0, alice}], @eth, [{alice, 10}])
+    {:ok, %{blknum: deposit_blknum}} = Client.call(:submit, %{transaction: tx})
+
+    bad_tx = API.TestHelper.create_recovered([{deposit_blknum, 0, 0, alice}], @eth, [{alice, 10}])
+
+    %{hash: bad_block_hash, number: bad_block_number, transactions: _} =
```

Comment:
same comment here on bad block number.

Also why is the bad block 27 not 40 here?

Reply:
1) `transaction which is using already spent utxo from exit and happened after margin of slow validator(m_sv) causes to emit invalid_block event`:
I put  40_000 because tx happened after m_sv.
2 `transaction which is using already spent utxo from exit and happened before end of margin of slow validator (m_sv) causes to emit invalid_exit event `
I put  27_000 because tx happened before m_sv.

As I look now on this I think I can calculate dynamically 40_000 and 27_000 based on `deposit_blknum` in this tests.